rust/connlib/tunnel/src/device_channel/tun_linux.rs

```diff
@@ -231,8 +237,30 @@ async fn set_iface_config(config: InterfaceConfig, _: Vec<IpAddr>, handle: Handl
         .await;
 
     handle.link().set(index).up().execute().await?;
-
     res_v4.or(res_v6)?;
+
+    // TODO: Gateways shouldn't set up DNS, right? Only clients?
+    if !dns_config.is_empty() {
+        // TODO: Before merging the DNS code to main, check if this would be better off
+        // in `on_set_interface_config` so it's handled by the client's CLI arg system
+        match connlib_shared::linux::get_dns_control_from_env() {
+            None => {
+                tracing::info!("Will not modify the system's DNS settings");
+            }
+            Some(DnsControlMethod::EtcResolvConf) => {
+                // TODO: Modify `/etc/resolv.conf`
+                tracing::info!("Will modify `/etc/resolv.conf`");
+            }
+            Some(DnsControlMethod::NetworkManager) => {
+                // TODO: Cooperate with NetworkManager
+            }
+            Some(DnsControlMethod::Systemd) => {
+                // TODO: Cooperate with `systemd-resolved`
+                tracing::info!("Will use `systemd-resolved`");
+            }
+        }
+    }
```

Comment:
I am happy to put this here if we agree that this is technical debt that we should fix ASAP once all the `async`, `Arc`, `Mutex` and worker cruft is removed from connlib.

Reply:
How far up the call-chain can we push this before it gets really ugly with interface changes? I think I'd prefer it to be ugly so it serves as a reminder that we need to fix it.

Reply:
Wherever we end up putting it, lets record an issue to capture the tech-debt.

Reply:
Agreed. I opened an issue here for debt for both this and the systemd PR https://github.com/firezone/firezone/issues/3636

In the latest commit I moved it up to the `new` function so it's pretty ugly. Technically if the FFI changed we could pass it through there, but it'll have to be removed again if we can get the `on_set_interface_config` callback or any other callback to set up DNS in the client.